### PR TITLE
Added default comment sort option.

### LINKED ISSUE #43: Default comment sort
Now that comment sort is implemented, there should be the ability to set a default comment sort.

### PR DESCRIPTION
**Pull request by Faalangst26**
Added the ability to set a default sort type for comments in the appearance menu, as it was a personal pet peeve of mine to have comments sorted to hot by default. 

resolves #43

### CODE REVIEW

**Comment by Faalangst26** [comment]
resolved merge conflicts

File: src/features/settings/appearance/CollapsedByDefault.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export default function CollapsedByDefault() {
   const dispatch = useAppDispatch();
   const { collapseCommentThreads } = useAppSelector(
     // this needs a better naming
```
Comment (rsammelson) [inline src/features/settings/appearance/CollapsedByDefault.tsx:20]: @aeharding Can this be removed because the item is named better?

File: src/features/settings/appearance/appearanceSlice.tsx [hunk 2]
```diff
@@ -36,7 +37,17 @@ export const OCommentThreadCollapse = {
 } as const;
 
 export type CommentThreadCollapse =
-  (typeof OCommentThreadCollapse)[keyof typeof OCommentThreadCollapse];
+    (typeof OCommentThreadCollapse)[keyof typeof OCommentThreadCollapse];
+
+export const OCommentDefaultSort = {
+  Hot: "Hot",
+  Top: "Top",
```
Comment (rsammelson) [inline src/features/settings/appearance/appearanceSlice.tsx:44]: I think it's conventional in this file for these strings (but not the identifiers) to be lowercase. The case is transformed before being displayed to the user.

File: src/features/settings/appearance/PostView.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ export default function PostView() {
   return (
     <>
       <ListHeader>
-        <IonLabel>Posts</IonLabel>
+        <IonLabel>Posts {typeof postsAppearanceType}</IonLabel>
```
Comment (rsammelson) [inline src/features/settings/appearance/PostView.tsx:15]: Why is this changed?

**Review by rsammelson — COMMENTED** [review]
This is a great feature

File: src/features/settings/appearance/PostView.tsx (hunk 3 above)
Comment (Faalangst26) [reply src/features/settings/appearance/PostView.tsx:15]: Good catch, left it in whilst debugging, Will remove and fix, the other things you mentioned. Thanks!

**Review by rsammelson — APPROVED** [review]

**Comment by rsammelson** [comment]
This fixes #43, right?

**Comment by Faalangst26** [comment]
resolved merge conflicts

**Comment by aeharding** [comment]
Thanks @Faalangst26 for keeping this up to date with main.

We will hopefully get the storage stuff in tomorrow (🤞) and then unblock this

**Comment by aeharding** [comment]
@Faalangst26 The DB work is done, so this should be ready to merge once conflicts resolved and updated to use the new db service! :)

Let me know if you have time in the next few days! Thank you!

**Comment by Faalangst26** [comment]
I'm on holiday at the moment, so I won't be able to resolve until I get back. You can close it if you want someone else to implement it in the meantime.

**Comment by rsammelson** [comment]
I can take over resolving the conflicts.

**Comment by Faalangst26** [comment]
@rsammelson I could merge it with this one if you open a PR with my branch. Alternatively you could just open a new PR on this repo and mention this one so it can get merged with main.

**Comment by rsammelson** [comment]
@Faalangst26, done

**Comment by aeharding** [comment]
Thanks for the pr. I'd like to get this in tomorrow if you have time.

Just checked - The appearance settings page needs some work. Tapping one of the options blanks out the page and the options are grouped strangely.

Also, sort isn't really an appearance setting. We should really make a general settings page. However, I'm fine merging this once the above gets resolved and we can fix this as part of a larger settings page refactor :)

**Comment by aeharding** [comment]
I've resolved conflicts, moved the setting to a new general page, and refactored a few things.

This should now be pretty much ready. If anyone want to test that would be appreciated.
